group/assignment6/RearrrangingCars.java

```diff
@@ -1,71 +1,85 @@
 package assignment_6;
 
+import java.util.HashSet;
 import java.util.Hashtable;
 import java.util.LinkedList;
 import java.util.Map.Entry;
-import java.util.Random;
 
 public class RearrrangingCars {
 
 	private Hashtable<Integer, Character> start;
 	private Hashtable<Integer, Character> end;
-	private Random rand;
-	private int noCarKey = 0;
-	private int noExistKey = -1;
+	private HashSet<Integer> carsInRightSlots;
+	private final int noCarKey = 0;
 
 	public RearrrangingCars(Hashtable<Integer, Character> start,
 			Hashtable<Integer, Character> end) {
-		rand = new Random();
+		carsInRightSlots = new HashSet<>();

```

Comment:
We can always check in constant time if a car is in the right slot rather than having to maintain a set of cars in their right slots..